pkg/util/syncutil/singleflight/singleflight.go

```diff
@@ -70,10 +70,25 @@ func (g *Group) Do(
 	return c.val, c.dups > 0, c.err
 }
 
-// DoChan is like Do but returns a channel that will receive the
-// results when they are ready. The method also returns a boolean
-// specifying whether the caller's fn function will be called or
-// not.
+// DoChan is like Do but returns a channel that will receive the results when
+// they are ready. The method also returns a boolean specifying whether the
+// caller's fn function will be called or not. This return value lets callers
+// identify a unique "leader" for a flight.
+//
+// NOTE: DoChan makes it possible to initiate or join a flight while holding a
```

Comment:
Consider pointing to `RangeCache.LookupRangeDescriptor` as an example of this pattern.